storage/database/dynamodb.go

```diff
@@ -347,7 +346,12 @@ func (dynamo *dynamoDB) Delete(key []byte) error {
 }
 
 func (dynamo *dynamoDB) Close() {
-	close(dynamo.quitCh)
+	if dynamoOpenedDBNum > 0 {
```

Comment:
Do we need to consider to protect `dynamoOpenedDBNum` for a race condition?

Reply:
Neither `NewDynamoDB()` and `Close()` is called concurrently. Do you want to protect it, just in case?

Reply:
Yes. But it is very rare case. Just ignore it.
```
func (dbm *databaseManager) Close() {
	// If single DB, only close the first database.
	if dbm.config.SingleDB {
		dbm.dbs[0].Close()
		return
	}

	// If not single DB, close all databases.
	for _, db := range dbm.dbs {
		if db != nil {
			db.Close()
		}
	}
}
```